packages/flutter/lib/src/material/menu_style.dart

```diff
@@ -198,6 +199,17 @@ class MenuStyle with Diagnosticable {
   /// as much of itself as possible, possibly overlapping the parent button.
   final AlignmentGeometry? alignment;
 
+  /// Used to override the menu size's animation curve and duration.
+  ///
+  /// If [AnimationStyle.duration] is provided, it will be used to override
+  /// the menu size's animation duration. Otherwise, defaults to 500ms.
+  ///
+  /// If [AnimationStyle.curve] is provided, it will be used to override
+  /// the menu size's animation curve. Otherwise, defaults to [Curves.easeInOutCubicEmphasized].
+  ///
+  /// To disable the animation when menu is open/close, use [AnimationStyle.noAnimation].
+  final AnimationStyle? sizeAnimationStyle;
```

Comment:
Does it eliminate the possibility of using different animation for open and close? For example, (although not Material) I observed that the dropdown menus on macOS has no animation for open, but a quick fade out for close.

Reply:
Yes! For this PR we only add a `sizeAnimationStyle` property. I think we can have another opacityAnimationStyle in a following PR if it is needed.

Reply:
Can we add `opacityAnimationStyle` in this PR? The special case at L408 (`if (effectiveAnimationStyle == defaultAnimationStyle) `) just doesn't feel right.

Also I think this property might be better renamed to `heightAnimationStyle`, since we're hard-coding `SizeTransition`'s axis to be vertical - unless we might also support horizontal in the future (which I doubt.)